AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m

```diff
@@ -67,23 +67,29 @@ - (BOOL)saveLog:(id<MSLog>)log withGroupId:(NSString *)groupId flags:(MSFlags)fl
                                    targetKey ? [NSString stringWithFormat:@"'%@'", targetKey] : @"NULL", (unsigned int)persistenceFlags];
   }
   return [self executeQueryUsingBlock:^int(void *db) {
+           // Selecting logs with equal or lower priority and ordering by priority then age.
            int result = [MSDBStorage executeNonSelectionQuery:addLogQuery inOpenedDatabase:db];
 
            // If the database is full, delete logs until there is room to add the log.
            long countOfLogsDeleted = 0;
-           while (result == SQLITE_FULL) {
-             result = [MSLogDBStorage deleteOldestLogsWithCount:1 inOpenedDatabase:db];
+           NSArray<NSNumber *> *ids = [self logsWithPriorityLowerThan:persistenceFlags inOpenedDatabase:db];
+           NSUInteger index = 0;
+           while (result == SQLITE_FULL && index < [ids count]) {
+             result = [MSLogDBStorage deleteLogsFromDBWithColumnValues:@[ ids[index++] ] columnName:kMSIdColumnName inOpenedDatabase:db];
```

Comment:
Maybe it's just me, but having `@[ ids[index++] ]` is a bit too much nesting? What do y'all think?

Reply:
I don't mind that nesting but I don't have a strong optionion

Reply:
I don't mind either.

Reply:
If it's within formatter max width before splitting, then it should be fine.